File: Hackerearth/inverselist.cpp
#include<iostream>
#include<vector>
#include<algorithm>

using namespace std;

bool is_inverse(vector<int> vec)
{
	for(int i=0;i<vec.size();i++)
	{
		int x=vec[i];
		int y=vec[x-1];
		if(x != vec[y-1])
			return 0;
	}
	return 1;
}

int main(void)
{
	int test_cases;
	cin >> test_cases;
	vector<int> v;
	for(int i=0;i<test_cases;i++)
	{
		v.clear();
		int size;
		cin >> size;
		for(int i=0,x;i<size;i++)
		{
			cin >> x;
			v.push_back(x);
		}

		if(is_inverse(v))
			cout << "inverse" << endl;
		else
			cout << "not inverse" << endl;
	}
	return 0;
}

File: Hackerearth/inverselist.py
#!/usr/bin/python

def is_inverse(list):
	for i in range(len(list)):
		x = list[i]
		y = list[x-1]
		if list[y-1] != x :
			print "not inverse"
	print "inverse"	

test_cases = input()
for i in range(1,test_cases+1):
	size = input()
	str = raw_input()
	l = str.split()
	for i in range(len(l)):
		var = int (l[i])
		list[i] = var
	is_inverse(list)
